### PR TITLE
implements POST /query

### PR DESCRIPTION
**Pull request by brunocalza**
# Summary

Implements the `POST /query` spec defined at https://github.com/tablelandnetwork/docs/pull/188

Working example on `staging`

```bash
curl -v --json '{"statement": "SELECT * FROM system_txn_receipts LIMIT 1"}' https://staging.tableland.network/api/v1/query
```

cc @dtbuchholz 

# Context

We're adding support for `POST /query` so queries can be sent via body request without the need of encoding the URL.

# Implementation overview

Generates the code based on a new spec and implements the controller in a similar way to `GET /query`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -80,7 +80,7 @@ lint:
 .PHONY: lint
 
 # OpenAPI
-SPEC_URL=https://raw.githubusercontent.com/tablelandnetwork/docs/main/specs/validator/tableland-openapi-spec.yaml
+SPEC_URL=https://raw.githubusercontent.com/tablelandnetwork/docs/bcalza/postquery/specs/validator/tableland-openapi-spec.yaml
```
Comment (brunocalza) [inline Makefile:83]: points to the new spec. gotta change this before merge

File: internal/router/controllers/controller.go [hunk 2]
```diff
@@ -265,6 +270,65 @@ func (c *Controller) GetTableQuery(rw http.ResponseWriter, r *http.Request) {
 	_, _ = rw.Write(formatted)
 }
 
+// PostTableQuery handles the POST /query call.
+func (c *Controller) PostTableQuery(rw http.ResponseWriter, r *http.Request) {
```
Comment (brunocalza) [inline internal/router/controllers/controller.go:274]: new handler for  `POST /query`

File: internal/router/router.go [hunk 3]
```diff
@@ -61,6 +61,10 @@ func configureAPIV1Routes(
 			userCtrl.GetTableQuery,
 			[]mux.MiddlewareFunc{middlewares.WithLogging, rateLim},
 		},
+		"QueryByStatementPost": {
```
Comment (brunocalza) [inline internal/router/router.go:64]: adds new route

File: internal/router/router.go [hunk 4]
```diff
@@ -97,11 +101,30 @@ func configureAPIV1Routes(
...
-			pathTemplate,
-			endpoint.handler,
-			append([]mux.MiddlewareFunc{middlewares.OtelHTTP(routeName)}, endpoint.middlewares...)...,
-		)
+		methods, err := route.GetMethods()
+		if err != nil {
+			return fmt.Errorf("get method: %s", err)
+		}
+
+		for _, method := range methods {
```
Comment (brunocalza) [inline internal/router/router.go:109]: connects the route with the handler

File: internal/router/controllers/controller.go [hunk 5]
```diff
@@ -256,6 +256,11 @@ func (c *Controller) GetTableQuery(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	if len(formatted) == 0 {
```
Comment (brunocalza) [inline internal/router/controllers/controller.go:259]: here it is @sanderpick

**Review by sanderpick — APPROVED** [review]
Lgtm!

**Review by avichalp — APPROVED** [review]
looks good!

**Review by avichalp — APPROVED** [review]

**Pull request merged**